pyccel/ast/builtin_objects/list_functions.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+
+"""
+The List container has a number of built-in methods that are 
+always available.
+
+This module contains objects which describe these methods within Pyccel's AST.
+"""
+
+from pyccel.ast.internals import PyccelInternalFunction
+from pyccel.ast.datatypes import NativeHomogeneousList
+
+class ListPop(PyccelInternalFunction) :
+    """
+    Represents a call to the .pop() method.
+    
+    Represents a call to the .pop() method which
+    removes the item at the specified index. 
+    The method also returns the removed item.
+
+    Parameters
+    ----------
+    list_variable : TypedAstNode
+        The name of the list.
+
+    index_element : TypedAstNode
+        The current index value for the element to be popped.
+    """
+    __slots__ = ('_dtype','_precision', '_index','_list_variable')
+    _rank = 0
+    _order = None
+    _shape = None
+    _class_type = NativeHomogeneousList()
+    name = 'pop'
+    def __init__(self, list_variable, index_element=None):
+        if index_element:
+            super().__init__(list_variable, index_element)
+        else:
+            super().__init__(list_variable)
+        self._index = index_element
+        self._list_variable = list_variable
```

Comment:
See the [docs](https://github.com/pyccel/pyccel/blob/devel/developer_docs/ast_nodes.md#pyccel-internal-function)

> The arguments can be passed through in the constructor of the sub-class to let PyccelInternalFunction take care of saving the arguments, or the arguments can be saved with more useful names inside the class. However care must be taken to ensure that the argument is not saved inside the class twice.

Here the argument is saved twice